### PR TITLE
Update elastic/security artifacts

### PR DESCRIPTION
**Pull request by achuguy**
Update the artifacts, composable templates, component templates, pipelines, ilm, used in the `elastic/security` track to Elasticsearch 8.15.1.  Also deleting unused artifacts from the repo.

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
@achuguy Under `composable-logsdb` there are copies of some files that are specific for the `logsdb` benchmark. Can we make sure the changes are applied to those files too? Thank you.

**Comment by salvatore-campagna** [comment]
Question: does that logic generating the dataset needs adjustment?

**Comment by achuguy** [comment]
> Question: does that logic generating the dataset needs adjustment?

For generating a dataset I'll take endpoint data from a cluster with real endpoints.  There is room to adjust fields in the dataset depending on cardinality needed.

**Comment by achuguy** [comment]
> @achuguy Under `composable-logsdb` there are copies of some files that are specific for the `logsdb` benchmark. Can we make sure the changes are applied to those files too? Thank you.

@salvatore-campagna When I'm trying to run the security-track with track-param `"index_mode": "logsdb"` I get an error trying to create the index template:

```
{
  "error": {
    "root_cause": [
      {
        "type": "x_content_parse_exception",
        "reason": "[37:7] [data_stream_template] unknown field [failure_store]"
      }
    ],
    "type": "x_content_parse_exception",
    "reason": "[37:24] [index_template] failed to parse field [data_stream]",
    "caused_by": {
      "type": "x_content_parse_exception",
      "reason": "[37:7] [data_stream_template] unknown field [failure_store]"
    }
  },
  "status": 400
}
```

Is the `failure_store` parameter correct under `data_stream`?

**Comment by salvatore-campagna** [comment]
> > @achuguy Under `composable-logsdb` there are copies of some files that are specific for the `logsdb` benchmark. Can we make sure the changes are applied to those files too? Thank you.
> 
> @salvatore-campagna When I'm trying to run the security-track with track-param `"index_mode": "logsdb"` I get an error trying to create the index template:
> 
> Create composable index template
> 
> ```
> PUT _index_template/logs-endpoint.events-process-test
> {
>     "index_patterns": [
>       "logs-endpoint.events.process-*"
>     ],
>     "template": {
>       "settings": {},
>       "mappings": {
>         "_meta": {
>           "package": {
>             "name": "endpoint"
>           },
>           "managed_by": "fleet",
>           "managed": true
>         }
>       }
>     },
>     "composed_of": [
>       "logs@mappings",
>       "logs@settings",
>       "logs-endpoint.events.process@package",
>       "ecs@mappings",
>       ".fleet_globals-1",
>       ".fleet_agent_id_verification-1",
>       "track-shared-logsdb-mode"
>     ],
>     "priority": 200,
>     "_meta": {
>       "package": {
>         "name": "endpoint"
>       },
>       "managed_by": "fleet",
>       "managed": true
>     },
>     "data_stream": {
>       "hidden": false,
>       "allow_custom_routing": false,
>       "failure_store": false
>     }
>   }
> ```
> 
> Error
> 
> ```
> {
>   "error": {
>     "root_cause": [
>       {
>         "type": "x_content_parse_exception",
>         "reason": "[37:7] [data_stream_template] unknown field [failure_store]"
>       }
>     ],
>     "type": "x_content_parse_exception",
>     "reason": "[37:24] [index_template] failed to parse field [data_stream]",
>     "caused_by": {
>       "type": "x_content_parse_exception",
>       "reason": "[37:7] [data_stream_template] unknown field [failure_store]"
>     }
>   },
>   "status": 400
> }
> ```
> 
> Is the `failure_store` parameter correct under `data_stream`?

If you look at the copies under `composable-logsdb` there is a Jinja annotation that is selectively including the line
```
"data_stream": {
      "hidden": false,
      "allow_custom_routing": false
      {% if build_flavor != "serverless" %},
      "failure_store": false
      {% endif %}
    }
```
based on `build_flavor`. So, it is correct depending on `build_flavor`

**Comment by achuguy** [comment]
Looks like `failure_store` is in the 8.16.0-SNAPSHOT and not in 8.15.x so I've adjusted my testing.

File: elastic/security/templates/composable/logs-endpoint.events.file.json
Comment (gbanasiak) [inline elastic/security/templates/composable/logs-endpoint.events.file.json:31]: @salvatore-campagna Are you OK with removal of `track-shared-logsdb-mode` here? Today when index mode is `logsdb` we take composable template definition for `logs-endpoint` integration from `composable-logsdb` directory, not `composable` directory like this file here. Do we need `track-shared-logsdb-mode` component template in any of composable templates in `composable` directory?

File: elastic/security/track.json
Comment (gbanasiak) [inline elastic/security/track.json:102]: We need to keep this logic. Today we use `logsdb` index mode in internal benchmarks with `elastic/security` with the assumption that all the configuration is coming from Kibana, and we only overwrite the bare minimum to configure the index mode. cc @salvatore-campagna

File: elastic/security/track.json
Comment (gbanasiak) [inline elastic/security/track.json:138]: @salvatore-campagna Can you comment on removal conditional based on `skip_delete_component_template` ex `skip_fleet_globals`? This was originally introduced in https://github.com/elastic/rally-tracks/pull/644. I'm not sure we need this at all. We are either using a vanilla cluster where we should be able to delete those component templates freely, or we use a setup where Kibana is installing those, and then we shouldn't touch those by skipping selected track tasks.

**Review by gbanasiak — COMMENTED** [review]
I gave it a first pass. I may report more once I run the new track definition against a vanilla 8.15.1 cluster. I've labelled the PR with `backport-to-8.15` to trigger automatic backport once this one is merged.

File: elastic/security/templates/component/logs-endpoint.events.library@package.json
Comment (gbanasiak) [reply elastic/security/templates/component/logs-endpoint.events.library@package.json:9]: Same applies to other `logs-endpoint` component templates, will hide those to reduce repetition.

File: elastic/security/templates/composable/logs-endpoint.events.file.json
Comment (salvatore-campagna) [reply elastic/security/templates/composable/logs-endpoint.events.file.json:31]: The assumption here is that `track-shared-logsdb-mode` is not required by others like `.logs-endpoint.action.responses` because for those we rely on kibana. I would then just remove it and remove it also from `composed_of`.

File: elastic/security/track.json
Comment (salvatore-campagna) [reply elastic/security/track.json:138]: As far as I recall now we should be ok with deleting them (not skipping delete).

**Review by salvatore-campagna — APPROVED** [review]

**Comment by salvatore-campagna** [comment]
LGTM after applying changes commented by @gbanasiak

File: elastic/security/templates/composable/logs-endpoint.events.file.json
Comment (gbanasiak) [reply elastic/security/templates/composable/logs-endpoint.events.file.json:31]: Ok, so let's remove including `security-*` composable templates. Please also remove `track-shared-logsdb-mode` component template installation when not in `logsdb` index mode in `track.json`.

File: elastic/security/track.json
Comment (gbanasiak) [reply elastic/security/track.json:138]: Ok, so let's remove it entirely, including this from `track.json`:
```
{% set p_skip_delete_component_template = (skip_delete_component_template | default(false) ) %}
```

File: elastic/security/ilm/security.json
Comment (achuguy) [reply elastic/security/ilm/security.json:15]: Whoops, good catch.

File: elastic/security/pipelines/logs@default-pipeline.json
Comment (achuguy) [reply elastic/security/pipelines/logs@default-pipeline.json:5]: I'll trim these two artifacts, `logs` ILM and `logs@default-pipeline`

File: elastic/security/templates/component/logs-endpoint.events.library@package.json
Comment (achuguy) [reply elastic/security/templates/component/logs-endpoint.events.library@package.json:9]: Updated to restore this switch

File: elastic/security/track.json
Comment (achuguy) [reply elastic/security/track.json:102]: Restored this logic.

File: elastic/security/templates/composable/logs-endpoint.events.file.json
Comment (achuguy) [reply elastic/security/templates/composable/logs-endpoint.events.file.json:31]: Done

**Comment by gbanasiak** [comment]
@achuguy I've added https://github.com/elastic/rally-tracks/pull/675/commits/a92162e6bf7a84b25dd2a1902009dd7ecf5b8b0c to avoid collision between packages required by Rally, and packages installed by the track.

**Comment by gbanasiak** [comment]
@salvatore-campagna 

> Looks like `failure_store` is in the 8.16.0-SNAPSHOT and not in 8.15.x so I've adjusted my testing.

We can't backport it to `8.15` with `failure_store`, this will need adjustment.

**Comment by pquentin** [comment]
To be clear, this is not a 8.16 vs 8.15 thing. `failure_store` is [only enabled in SNAPSHOT builds](https://github.com/elastic/elasticsearch/blob/b8b5a92de3a742dc060ba3b60b49dce155a97322/modules/data-streams/build.gradle#L38-L42) and should go away completely at some point.

**Comment by gbanasiak** [comment]
Now with `logs-endpoint` composable templates being up-to-date we can simplify further by removing `composable-logsdb` directory. I'm also decoupling index mode setting from exclusions for benchmark scenarios with Kibana by introducing additional `logs_endpoint_from_kibana` track parameter. See https://github.com/elastic/rally-tracks/pull/675/commits/9e1089b75e5f334aa5fe3184425851a9da10d7c0.

**Comment by gbanasiak** [comment]
https://github.com/elastic/rally-tracks/pull/675/commits/2bafd52864eb86d6ba75539a55a24e9b143f82eb moves the track closer to being fully supported in serverless (both standard user and operator). What remains to be done on that front (outside of this PR) is similar adjustment of Beats composable templates and addition of serverless IT tests ([here](https://github.com/elastic/rally-tracks/tree/master/it_serverless)).

**Comment by salvatore-campagna** [comment]
> To be clear, this is not a 8.16 vs 8.15 thing. `failure_store` is [only enabled in SNAPSHOT builds](https://github.com/elastic/elasticsearch/blob/b8b5a92de3a742dc060ba3b60b49dce155a97322/modules/data-streams/build.gradle#L38-L42) and should go away completely at some point.

Thi si because it is using a feature flag right? That at some point will be removed.

**Review by salvatore-campagna — APPROVED** [review]

**Comment by salvatore-campagna** [comment]
Thanks @gbanasiak and @achuguy for fixing this!

File: elastic/security/README.md
Comment (gbanasiak) [inline elastic/security/README.md:105]: Index mode is already described above, that's a duplicate.

**Review by gbanasiak — APPROVED** [review]

**Comment by gbanasiak** [comment]
@achuguy Can I ask for a manual backport to `8.15`?

**Pull request merged**
